connector/src/main/scala/com/microsoft/kusto/spark/datasink/KustoWriter.scala

```diff
@@ -231,21 +234,20 @@ object KustoWriter {
                                                batchIdForTracing: String,
                                                rows: Iterator[InternalRow],
                                                partitionsResults: CollectionAccumulator[PartitionResult],
-                                               parameters:KustoWriteResource): Unit = {
+                                               parameters: KustoWriteResource): Unit = {
 
     val partitionId = TaskContext.getPartitionId
-    KDSU.logInfo(myName, s"Processing partition: '$partitionId' in requestId: '${parameters.writeOptions.
-      requestId}'$batchIdForTracing")
+    KDSU.logInfo(myName, s"Processing partition: '$partitionId' in requestId: '${
+      parameters.writeOptions.
+        requestId
+    }'$batchIdForTracing")
     val ingestClient = KustoClientCache.getClient(parameters.coordinates.clusterUrl,
       parameters.authentication, parameters.coordinates.ingestionUrl, parameters.coordinates.clusterAlias).ingestClient
-    val queueRequestOptions = new QueueRequestOptions
-    queueRequestOptions.setMaximumExecutionTimeInMs(KCONST.DefaultExecutionQueueing)
-    queueRequestOptions.setTimeoutIntervalInMs(KCONST.DefaultTimeoutQueueing)
-    queueRequestOptions.setRetryPolicyFactory(new RetryNoRetry)
-    ingestClient.setQueueRequestOptions(queueRequestOptions)
+    val reqRetryOpts = new RequestRetryOptions(RetryPolicyType.EXPONENTIAL, Integer(KCONST.MaxIngestRetryAttempts), Duration.ofSeconds(KCONST.DefaultMaximumIngestionTime.toSeconds), null, null, null)
```

Comment:
This is very different then what was before!
The reason we had NoRetry here is that we favor longer timeout over retries when it comes to queueing or else we get duplications

Reply:
You can do something similar to .Net
https://msazure.visualstudio.com/One/_git/Azure-Kusto-Service?path=/Src/Client/Kusto.Ingest/Storage/QueueOptions.cs

Reply:
Not fixed